docs/enhancements/support-node-labeling.md

```diff
@@ -0,0 +1,128 @@
+---
+title: node labeling support
+authors:
+- "@oamizur"
+  creation-date: 2022-12-20
+  last-updated: 2022-12-20
+---
+
+# Support node labeling during installation
+
+## Summary
+
+There was a requirement to support node roles during installation.  These roles 
+are extra roles, which are used by the user and not by any Openshift components.
+These labels should be applied on agent CR which represents a node in a spoke cluster.
+
+## Motivation
+
+Currently, nodes can be assigned with a single role which can be one of two values [ master, worker].
+The may want to add additional roles.  These roles are for the user usage only.
+
+### Goals
+- For ZTP flow: add capability to add and update labels on BMH which will be transferred 
+to agent CR.
+- For boot it yourself flow: add capability to add and update labels to agent CR after agent
+  has been created.
+
+### Non goals
+- Single way to set node labels before agent creation both for ZTP flow, and boot it yourself flow.
+- REST API support.  Node labeling will not be supported under REST API.
+
+## Proposal
+
+There are 2 implementation options: Using labels, or adding dedicated spec field.
+
+### Using labels
+Labels can be applied on every CR.  So there is no known limitation to add a label on agent.
+
+For the ZTP flow, there should a pattern for transferring an annotation from the BMH to a label  
+in the corresponding agent.  The pattern is actually the prefix of the annotation.
+It should be **agent-install.openshift.io.user.role**.  These annotations will be transferred to 
+to labels on the corresponding agent CR.
+
+### Adding dedicated spec field in agent CR
+Additional field **userRoles** will be added to the spec 
+section of the agent CR.  This field will be a string list.
+
+For the ZTP flow, the annotations having the format **agent-install.openshift.io.user.role/[key]=[value]**,
+the value part will be transferred to the **userRoles** field as additional entry. 
+
+### User Stories
+
+#### Story 1
+
+Let the user define roles specific to the business domain in addition to 
```

Comment:
what will it transfer to in OCP? what will it affect?

Reply:
It will be transferred to labels on nodes

Reply:
to labels on nodes.
It may effect the user role which may cause attaching to different MCP